sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts

```diff
@@ -92,9 +92,39 @@ export abstract class SFTPIntegrationBase<DownloadType> {
    */
   protected async downloadResponseFileLines(
     remoteFilePath: string,
-  ): Promise<string[]> {
+  ): Promise<string[]>;
+
+  /**
+   * Downloads the file specified on 'fileName' parameter from the
+   * SFAS integration folder on the SFTP.
+   * @param remoteFilePath full remote file path with file name.
+   * @param checkIfFileExist when set to true, check if file
+   * exist before downloading it.
+   * @returns parsed records from the file.
+   */
+  protected async downloadResponseFileLines(
+    remoteFilePath: string,
+    checkIfFileExist: boolean,
+  ): Promise<string[] | false>;
+
+  /**
+   * Downloads the file specified on 'fileName' parameter from the
+   * SFAS integration folder on the SFTP.
+   * @param remoteFilePath full remote file path with file name.
+   * @returns parsed records from the file.
+   */
+  protected async downloadResponseFileLines(
+    remoteFilePath: string,
+    checkIfFileExist = false,
```

Comment:
Can we have it converted to the `options` format?

Reply:
also, pls add the comment